docs/configuration/config-file-reference.md

```diff
@@ -659,6 +659,19 @@ store_gateway_client:
   # TLS CA path for the client
   # CLI flag: -experimental.querier.store-gateway-client.tls-ca-path
   [tls_ca_path: <string> | default = ""]
+
+# Configures the consistency check done by the querier on queried blocks when
+# running the experimental blocks storage.
+blocks_consistency_check:
+  # Whether the querier should run a consistency check to ensure all expected
+  # blocks have been queried.
+  # CLI flag: -experimental.querier.blocks-consistency-check.enabled
+  [enabled: <boolean> | default = true]
+
+  # The grace period allowed before a new block is included in the consistency
+  # check.
+  # CLI flag: -experimental.querier.blocks-consistency-check.upload-grace-period
+  [upload_grace_period: <duration> | default = 1h]
```

Comment:
The default seems excessively high?

Reply:
Unfortunately we have to protect from the compactor as well, and the block ID is generated at the beginning of the compaction so it could easily pass 1h. However, this `upload_grace_period` will soon disappear (it's a temporarily hack) as soon as we have this done (I'm going to work on it right now): https://github.com/thanos-io/thanos/issues/2585